### ISSUE TITLE
Artist/Album Sorting Issue

### ISSUE THREAD

**Issue by BigBlakBoi** [timusus/Shuttle#334]
**Shuttle version:**
 
 v2.0.4
 
**Device, OS:**
 
OnePlus 5, Android 8.1.0

**Description of bug:**   

 A large number of artists are missing from the artist tab, and instead are all stored under one artist's name.What those songs have in common is they have no album name, and therefore are given the default album name "music". This is identical to issue #52  which is now closed, but the problem still exists. Looking at issue #52 should recap exactly what the problem is.
 
**Steps to reproduce:**
 
 1. Store music in "music" folder without an album name
 2.  Open the app and go to artist or album tab
 
**Expected outcome:**
 
Songs will be listed under their given artist as per their ID3 artist tag.
 
**Observations/Actual Result:**
 
 In the artist tab, songs without an album will be stored under the same artist or "<unknown>"
![issue01](https://user-images.githubusercontent.com/26145281/39585978-fbe4565e-4ec3-11e8-9455-d51832dbf390.jpg)
![issue02](https://user-images.githubusercontent.com/26145281/39585979-fbfab0c0-4ec3-11e8-9127-0fc2e60c6d16.jpg)

**Comment by timusus** [comment]
It looks to me like the issue is there is no artist/album-artist information in the tags.

**Comment by BigBlakBoi** [comment]
@timusus for some reason you're correct, the artist names aren't there (they were there yesterday which is weird). But just now I edited the tags and put in the artists for some of the songs. They still do not leave the "music" album and remain inside the "unknown" artist. The problem is not contingent on the artist name but rather the album name being prioritized over the artist name.

**Comment by timusus** [comment]
It sounds to me like you just need to fix up your tags. If you've correctly entered the artist name, or album artist names, then songs belonging to that artist/album-artist should no longer be grouped under the same album as tags without artist/album-artist information, that's the fix that was made in #52. 

Most likely, you've edited your tags but those changes haven't yet propagated to the Media Store. You can try to fix that by 'rescanning' the library after editing the tags. You can rescan a folder via the overflow menu in the folder browser (Shuttle+ only). 

There are unit tests in Shuttle which cover these cases, and they appear to be passing, so I'm fairly confident this is an issue you can address on your end.

**Comment by BigBlakBoi** [comment]
@timusus you're correct but it's technically still a problem. When the album-artist tag is left empty, this problem still happens, regardless of whether or not there is a present artist tag. It completely ignores the artist tag and looks for the album-artist tag, if it's not there, then this problem arises. This doesn't make sense for a single that has no album. How can there be an album artist if there is no album?

The problem can be "fixed" if you just fill in the album-artist, but that shouldn't be necessary in the first place. For now however I guess I'll just have to retag the album-artist with the same tag as the artist, which should be redundant work.

**Comment by timusus** [comment]
You're right, providing an album-artist shouldn't be required to work around this issue - providing the artist alone should be adequate. In fact, the album-artist is derived from the artist if album-artist is not found in the tags, so I'm surprised that manually entering an album-artist fixes this problem for you.

I'll look into it.

**Comment by BigBlakBoi** [comment]
I really appreciate you actually taking a look and not just blowing off my concern. Much appreciated.

**Comment by kikiotsuka** [comment]
Hi, any fix for this? I'm having the same problem where I don't have an album-artist and all my songs go into a random artist bundled in an album called "Music" exactly the same as the pictures shown in this thread

**Comment by BigBlakBoi** [comment]
To check I removed the album-artist name from a few songs to see if this problem would be recreated; it wasn't. As far as I'm concerned the problem is fixed. I recommend playing with the tags in songs to see if doing anything fixes it. Once you've figured out the cause then you can work from there.

**Comment by timusus** [comment]
@BigBlakBoi I haven't made any changes to the app to fix this problem.. Perhaps it only occurs if you have album-artist specified, but no album?

**Comment by kikiotsuka** [comment]
The songs that are incorrectly placed under the wrong artist all have a Title, a genre, an artist (but not the artist its bundled under) and a null value for my album. @BigBlakBoi did you clear the mediastore cache and force refresh it after removing labels?

**Comment by BigBlakBoi** [comment]
@kikiotsuka no I didn't, but to be honest I'm not really keen on trying to recreate this problem and getting myself in a position where I'm back with this problem again. Honestly, just like I said way earlier in the thread, try filling in the album-artist tag and seeing if that works. If it does it's likely the only work around at the moment.

**Comment by BigBlakBoi** [comment]
@kikiotsuka I can confirm the problem still exists. Shuttle is prioritizing album-artist over the artist. Even when I change the artist name, it stays in place unless I change the album-artist, then it will recognize the artist's name has changed. Can't tell you why this is the case though.

**Comment by blueorignal** [comment]
I finally figured out what was causing this. In each case the issue was having an embedded picture in the music file that was very large (i.e. 2kx2k or 4kx4k pixels). Also some of the pictures were not exactly square i.e. 3500x3497 pixels.

To fix the issue I completed the following steps for the effected files:
- Exported the existing embedded image from the file concerned
- Squared the image (made each side the exact same number of pixels)
- Resized the image to 1000x1000 pixels
- Deleted the original embedded images and updated the music files to have no image
- Embedded the new image file into the music files and saved the changes

By replacing these files I was able to resolve the issue of the unknown artist (despite the artist and album-artist tags being correct).

The program I used to edit the tags was kid3 Audio Tagger (https://kid3.kde.org/), however I suspect that any tag editor will work.

@timusus - I suspect the array used to read the image in your code is too small to support very large images, although this is just a guess as I have not done a deep dive into the code. Irrespective, hopefully the fix helps anyone else encountering this issue. Thanks again for your **amazing** player.
